pyspecdata/core.py

```diff
@@ -7282,32 +7237,33 @@ def fit(self,set = None, set_to = None, force_analytical = False, silent = False
-        try:
-            p_out,cov,infodict,mesg,success = leastsq(*leastsq_args,**leastsq_kwargs)
-        #{{{ just give various explicit errors
-        except TypeError as err:
-            if not isinstance(x, ndarray) and not isinstance(y, ndarray):
-                raise TypeError(strm('leastsq failed because the two arrays',
-                    "aren\'t of the right",
-                    'type','type(x):',type(x),'type(y):',type(y)))
-            else:
-                if any(shape(x) != shape(y)):
-                    raise RuntimeError(strm('leastsq failed because the two arrays do'
-                            "not match in size size",
-                            'shape(x):',shape(x),
-                            'shape(y):',shape(y)))
-            raise TypeError(strm('leastsq failed because of a type error!',
-                'type(x):',showtype(x),'type(y):',showtype(y),
-                'type(sigma)',showtype(sigma),'shape(x):',shape(x),
-                'shape(y):',shape(y),'shape(sigma)',shape(sigma),
-                'p_ini',type(p_ini),p_ini))
-        except ValueError as err:
-            raise ValueError(strm('leastsq failed with "',err,
-                '", maybe there is something wrong with the input:',
-                self))
-        except Exception as e:
-            raise ValueError('leastsq failed; I don\'t know why'+explain_error(e))
-        #}}}
+        p_out,cov,infodict,mesg,success = leastsq(*leastsq_args,**leastsq_kwargs)
+        #try:
+        #   p_out,cov,infodict,mesg,success = leastsq(*leastsq_args,**leastsq_kwargs)
+        ##{{{ just give various explicit errors
+        #except TypeError as err:
+        #    if not isinstance(x, ndarray) and not isinstance(y, ndarray):
+        #        raise TypeError(strm('leastsq failed because the two arrays',
+        #            "aren\'t of the right",
+        #            'type','type(x):',type(x),'type(y):',type(y)))
+        #    else:
+        #        if any(shape(x) != shape(y)):
+        #            raise RuntimeError(strm('leastsq failed because the two arrays do'
+        #                    "not match in size size",
+        #                    'shape(x):',shape(x),
+        #                    'shape(y):',shape(y)))
+        #    raise TypeError(strm('leastsq failed because of a type error!',
+        #        'type(x):',showtype(x),'type(y):',showtype(y),
+        #        'type(sigma)',showtype(sigma),'shape(x):',shape(x),
+        #        'shape(y):',shape(y),'shape(sigma)',shape(sigma),
+        #        'p_ini',type(p_ini),p_ini))
+        #except ValueError as err:
+        #    raise ValueError(strm('leastsq failed with "',err,
+        #        '", maybe there is something wrong with the input:',
+        #        self))
+        #except Exception as e:
+        #    raise ValueError('leastsq failed; I don\'t know why'+explain_error(e))
+        ##}}}
```

Comment:
does it run if you uncomment these? if so I would leave this in

Reply:
I uncommented these and it did run, so I am leaving them